lib/parser.ex

```diff
@@ -0,0 +1,50 @@
+defmodule DateTimeParser.Parser do
+  @moduledoc false
+  alias DateTimeParser.{Epoch, Serial}
+
+  @epoch_regex ~r|\A(?<seconds>\d{10,11})(?:\.(?<subseconds>\d{1,10}))?\z|
+  @serial_regex ~r|\A-?\d{1,5}(?:\.\d{1,10})?\z|
+  @default_parsers [:epoch, :serial, :tokenizer]
+
+  def get_parser(string, opts) do
+    opts = Keyword.merge([tokenizer: get_tokenizer(opts[:context], string)], opts)
+
+    opts
+    |> Keyword.get(:parsers, Application.get_env(DateTimeParser, :parsers, @default_parsers))
+    |> Enum.find_value({:error, :no_parser}, fn parser ->
```

Comment:
Does setting `{:error, :no_parser}` as the default mean if nothing matches that is what is returned (and presumably printed in an exception message somewhere)?

Reply:
The error messaging doesn't bubble up to the user, since a `with` captures all non `:ok` tuples and returns a generic error message, but it does help with debugging.